File: languages/java/Prime.java
// languages/java/Prime.java
import java.util.Scanner;

public class Prime {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a number to check if it's prime: ");
        int n = scanner.nextInt();
        scanner.close();

        if (isPrime(n)) {
            System.out.println(n + " is a prime number.");
        } else {
            System.out.println(n + " is not a prime number.");
        }
    }

    public static boolean isPrime(int n) {
        if (n <= 1) return false;
        if (n <= 3) return true;
        if (n % 2 == 0 || n % 3 == 0) return false;
        for (int i = 5; i * i <= n; i += 6) {
            if (n % i == 0 || n % (i + 2) == 0) return false;
        }
        return true;
    }
}

File: languages/perl/prime.pl
#!/usr/bin/perl
use strict;
use warnings;
use POSIX qw(floor);

sub is_prime {
    my ($n) = @_;
    
    return 0 if $n <= 1;
    return 1 if $n == 2;
    return 0 if $n % 2 == 0;
    
    for (my $i = 3; $i <= floor(sqrt($n)); $i += 2) {
        return 0 if $n % $i == 0;
    }
    return 1;
}

print "Enter a number to check if it's prime: ";
my $number = <STDIN>;
chomp($number);

if (is_prime($number)) {
    print "$number is a prime number.\n";
} else {
    print "$number is not a prime number.\n";
}
